### PR TITLE
Renamed special-resource-operator --> kernel-module-management.

### PR DESCRIPTION
**Pull request by ybettan**
Also removed some outdated tests and bumped golang version to 1.18

Signed-off-by: Yoni Bettan <yonibettan@gmail.com>

### CODE REVIEW

**Comment by ybettan** [comment]
FYI @qbarrand

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml [hunk 1]
```diff
@@ -1,41 +1,5 @@
 presubmits:
   kubernetes-sigs/special-resource-operator:
```
Comment (qbarrand) [inline config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:2]: We probably need to change all references to SRO as well.

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml [hunk 2]
```diff
@@ -45,9 +9,9 @@ presubmits:
       description: "lint the source code of special-resource-operator."
     spec:
       containers:
-      - image: golang:1.17
+      - image: golang:1.18
         command:
-        - scripts/test-infra/lint.sh
+        - ci/prow/lint.sh
```
Comment (qbarrand) [inline config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:14]: WDYT about removing the extension? The [Google style guide](https://google.github.io/styleguide/shellguide.html#file-extensions) strongly recommends no extension for executables.

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml [hunk 3]
```diff
@@ -57,18 +21,6 @@ presubmits:
...
-      testgrid-dashboards: sig-node-special-resource-operator
-      testgrid-tab-name: e2e-k8s
-      description: "e2e-k8s the source code of special-resource-operator."
-    spec:
-      containers:
-      - image: golang:1.17
+      - image: golang:1.18
         command:
-        - scripts/test-infra/e2e_k8s.sh
+        - ci/prow/unit_tests.sh
```
Comment (qbarrand) [inline config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:26]: I would remove the extension here as well.

**Comment by ybettan** [comment]
@mrunalp Can we please have you review here and the `ok-to-test` label?

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml (hunk 1 above)
Comment (ybettan) [reply config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:2]: Good catch

Comment (ybettan) [reply config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:2]: We should probably leave it that way until the repo is renamed though.
I can add a task to change the names once the repo has been renamed if that's OK.

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml (hunk 2 above)
Comment (ybettan) [reply config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:14]: I do like the `.sh` extension. It lets you know that this is a bash script and not just a compiled binary.
Also, no crucial, but this also aligned with what we will soon have with NFD - https://github.com/kubernetes-sigs/node-feature-discovery/tree/master/scripts/test-infra

Comment (ybettan) [reply config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml]: Fixed.

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml (hunk 3 above)
Comment (ybettan) [reply config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml]: Fixed.

File: config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml [hunk 4]
```diff
@@ -57,18 +21,6 @@ presubmits:
...
-      testgrid-dashboards: sig-node-special-resource-operator
-      testgrid-tab-name: e2e-k8s
-      description: "e2e-k8s the source code of special-resource-operator."
-    spec:
-      containers:
-      - image: golang:1.17
+      - image: golang:1.18
         command:
-        - scripts/test-infra/e2e_k8s.sh
+        - ci/prow/unit_tests
```
Comment (qbarrand) [inline config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:26]: If we want to match the `Makefile` target:
```suggestion
        - ci/prow/unit-test
```

Comment (ybettan) [reply config/jobs/kubernetes-sigs/special-resource-operator/special-resource-operator-presubmits.yaml:26]: This is not the script name, see https://github.com/qbarrand/oot-operator/blob/main/ci/prow/unit_tests.

IMO, the convention should be as follow:
* Prow refers to bash scripts in the `ci/prow/file_name` (example `ci/prow/unit_tests`)
* Those scripts can call a single Makefile target (such as `make unit-tests`) or can contains multiple instructions (such as e2e script will do)

**Comment by ybettan** [comment]
Relates to issue https://github.com/kubernetes/org/issues/3587

**Comment by ybettan** [comment]
@dims Is there anything needed that is preventing us from `lgtm`?

**Comment by dims** [comment]
/approve
/lgtm

@ybettan in the future, you can ping the folks listed in the OWNERS file where you updated a file(s). For example here, you would ping:
https://github.com/kubernetes/test-infra/blob/28c7fea0d3c3c5a05a839c85db7b22049ddeb7f9/config/jobs/kubernetes-sigs/kernel-module-management/OWNERS#L4-L6

**Comment by ybettan** [comment]
TY @dims .
Will ping the correct people next time. Sorry for the noise.

**Pull request merged**